src/Rbac/DefaultRoleManager/RoleManager.php

```diff
@@ -351,4 +313,20 @@ protected function &loadOrStoreRoles(string $domain, Roles $roles): Roles
 
         return $this->allDomains[$domain];
     }
-}
+
+    /**
+     * @param string $domain
+     *
+     * @return Roles
+     */
+    protected function &checkHasDomainPatternOrHasPattern(string $domain): Roles
+    {
+        if ($this->hasDomainPattern || $this->hasPattern) {
+            $allRoles = &$this->generateTempRoles($domain);
+        } else {
+            $allRoles = &$this->loadOrStoreRoles($domain, new Roles());
+        }
+
+        return $allRoles;
+    }
+}
```

Comment:
```suggestion
/**
 * Retrieves roles based on domain patterns.
 *
 * @param string $domain
 * @return Roles
 */
protected function &getRolesBasedOnDomainPattern(string $domain): Roles
{
    if ($this->shouldGenerateTempRoles()) {
        return $this->generateTempRoles($domain);
    }

    return $this->loadOrStoreRoles($domain, new Roles());
}

/**
 * Determines if temporary roles should be generated based on domain patterns.
 *
 * @return bool
 */
private function shouldGenerateTempRoles(): bool
{
    return $this->hasDomainPattern || $this->hasPattern;
}
```

Like previously seen the namming should be changed to be more revelant about his intentions.
Extract the conditional logic in proper method
IMHO references are unecessary, to be double checked.
Small clarification about DOCbloc (It's mandatory in constributed context).